qa-tests-backend/src/main/groovy/services/ClusterService.groovy

```diff
@@ -30,6 +31,12 @@ class ClusterService extends BaseService {
         return getClusterServiceClient().getCluster(Common.ResourceByID.newBuilder().setId(clusterId).build()).cluster
     }
 
+    @Retry(attempts = 30, delay = 3)
+    static void waitForClusterHealthy() {
+        def status = getCluster().healthStatus
+        assert status.overallHealthStatus == ClusterOuterClass.ClusterHealthStatus.HealthStatusLabel.HEALTHY
```

Comment:
It would be good to check what exactly defines `overallHealthStatus`.

My concern is that maybe we will have an unhealthy overall state for some combination of tests. i.e. version compatibility tests, different platforms (AKS, etc.) - and such cases we are not able to catch in PRs. If we could run "main" branch merge commit tests in PR, it would be good to validate that.

Another approach would be: merge and check everything on merge commit 😈